app/services/marketplace_service/api/marketplaces.rb

```diff
@@ -148,6 +204,10 @@ def create_community_customization!(community, marketplace_name, locale)
         community.community_customizations.create(customization_params(marketplace_name, locale))
       end
 
+      def first_or_create_community_customization!(community, marketplace_name, locale)
+        community.community_customizations.first_or_create(customization_params(marketplace_name, locale))
```

Comment:
Did you test this "first_or_create" function?
Also, there is a mismatch with using "!" -> you should use the same behaviour inside your function as in its definition.